core/iwasm/aot/aot_runtime.c

```diff
@@ -1765,7 +1765,7 @@ aot_instantiate(AOTModule *module, AOTModuleInstance *parent,
 
 #if WASM_ENABLE_PERF_PROFILING != 0
     total_size = (uint64)sizeof(AOTFuncPerfProfInfo)
-                 * (module->import_func_count + module->func_count);
+                 * ((uint64)module->import_func_count + module->func_count);
```

Comment:
what problem does this cast solve?

Reply:
The compiler hints that integer overflow may occur in `module->import_func_count + module->func_count`.

Reply:
but we know this doesn't overflow, don't we?
i wonder what made the compiler think this particular addition may be more problematic than others.

Reply:
Here it should not overflow, the overflow was checked in aot loader:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/aot/aot_loader.c#L2487
I just changed the code to let the compiler happy.

Maybe it is due to the adding result is converted to uint64. The msvc reported:
```
Warning	C26451	Arithmetic overflow: Using operator '+' on a 4 byte value and
then casting the result to a 8 byte value. Cast the value to the wider type before
calling operator '+' to avoid overflow (io.2).
```